### ISSUE TITLE
Latin.traineddata(best) - Words missing in OCR

### ISSUE THREAD

**Issue by TheSeiko** [tesseract-ocr/tesseract#1080]
### Environment

* **Tesseract Version**: tesseract-ocr-setup-4.0.0-alpha.20170804.exe - Latin.traineddata from best
* **Platform**: Win10 64bit

### Current Behavior:
Tesseract skippes words when doing OCR

OCR-Result: USA

### Expected Behavior:

OCR-Result: USA/NORDKOREA

### Suggested Fix:

not to skip words

---
Today I saw multiple times, that tesseract skips words, sometimes in the middle of a paragraph.

i.e. tesseract test/1502433849760_1.png test/1502433849760 -l Latin

![1502433849760_imageprocessedwithmarks](https://user-images.githubusercontent.com/30631253/29271637-bbce3160-80fc-11e7-924e-bdc5f89cca34.png)

![1502433849760_1](https://user-images.githubusercontent.com/30631253/29271612-9ac83362-80fc-11e7-9e84-aa3c650f3618.png)
[1502433849760.txt](https://github.com/tesseract-ocr/tesseract/files/1222190/1502433849760.txt)

![1502433849760_screen](https://user-images.githubusercontent.com/30631253/29271704-0fb2feaa-80fd-11e7-87bc-97a53dc5c583.jpg)

**Comment by amitdo** [comment]
Known problem for all langs/scripts
https://github.com/tesseract-ocr/tesseract/issues/681

**Comment by amitdo** [comment]
In your case the problem is probaly in the layout analysis stage.

**Comment by TheSeiko** [comment]
I've another example:

![1502442621178_screen](https://user-images.githubusercontent.com/30631253/29272519-b46c3aee-8100-11e7-9ad3-734e3a1b46dc.jpg)

![1502442621178](https://user-images.githubusercontent.com/30631253/29272525-b9762536-8100-11e7-8d82-0961dd49663b.png)

[1502442621178.txt](https://github.com/tesseract-ocr/tesseract/files/1222269/1502442621178.txt)

**Comment by amitdo** [comment]
Try cutting the non-text areas with gimp and retest.

**Comment by TheSeiko** [comment]
Only the red areas are processed, they are written to separate png-files. (I've uploaded the separate files as well and the resulting output-files)

**Comment by TheSeiko** [comment]
The interesting thing is, two similar frames were processed without problems:

![1502434118849](https://user-images.githubusercontent.com/30631253/29273266-f7bbc636-8103-11e7-8efb-eb21a9e1fca6.jpg)

![1502431789149](https://user-images.githubusercontent.com/30631253/29273306-14a66bf2-8104-11e7-9844-a47bfab79d4c.jpg)

**Comment by amitdo** [comment]
>Here a word in the middle of the sentence is skipped.

Which word?

Might be related to #681 in this case.

**Comment by TheSeiko** [comment]
Pöllan -  the ö exists in unicharset

Thank you for the link to #681

**Comment by amitdo** [comment]
Try to debug with stopper_debug_level=2

https://github.com/tesseract-ocr/tesseract/blob/3ec11bd37a56/ccmain/linerec.cpp#L293

**Comment by TheSeiko** [comment]
Best choice certainty=-2.96366, space=-0.206939, scaled=-20.7456, final=-20.7456
 : Pöllan : R=11.9817, C=-2.96366, F=1, Perm=2, xht=[0,3.40282e+038], ambig=0
pos	NORM	NORM	NORM	NORM	NORM	NORM
str	P	ö	l	l	a	n
state:	1 	1 	1 	1 	1 	1 
C	-0.237	-2.964	-0.232	-0.209	-0.192	-0.199
Deleting word with certainty -20.7456
 : Pöllan : R=11.9817, C=-20.7456, F=1, Perm=2, xht=[0,3.40282e+038], ambig=0
pos	NORM	NORM	NORM	NORM	NORM	NORM
str	P	ö	l	l	a	n
state:	1 	1 	1 	1 	1 	1 
C	-0.237	-2.964	-0.232	-0.209	-0.192	-0.199

**Comment by TheSeiko** [comment]
Best choice certainty=-0.106016, space=-2.97242, scaled=-20.8069, final=-20.8069
 : USA : R=2.24336, C=-0.106016, F=1, Perm=8, xht=[0,3.40282e+038], ambig=0
pos	NORM	NORM	NORM
str	U	S	A
state:	1 	1 	1 
C	-0.090	-0.086	-0.106
Best choice certainty=-0.229416, space=-2.97242, scaled=-20.8069, final=-20.8069
 : /NORDKOREA : R=16.2872, C=-0.229416, F=1, Perm=2, xht=[0,3.40282e+038], ambig=0
pos	NORM	NORM	NORM	NORM	NORM	NORM	NORM	NORM	NORM	NORM
str	/	N	O	R	D	K	O	R	E	A
state:	1 	1 	1 	1 	1 	1 	1 	1 	1 	1 
C	-0.223	-0.202	-0.207	-0.201	-0.192	-0.208	-0.197	-0.193	-0.229	-0.228
Deleting word with certainty -20.8069
 : /NORDKOREA : R=16.2872, C=-20.8069, F=1, Perm=2, xht=[0,3.40282e+038], ambig=0
pos	NORM	NORM	NORM	NORM	NORM	NORM	NORM	NORM	NORM	NORM
str	/	N	O	R	D	K	O	R	E	A
state:	1 	1 	1 	1 	1 	1 	1 	1 	1 	1 
C	-0.223	-0.202	-0.207	-0.201	-0.192	-0.208	-0.197	-0.193	-0.229	-0.228

[output_1502433849760_1.txt](https://github.com/tesseract-ocr/tesseract/files/1222585/output_1502433849760_1.txt)

**Comment by amitdo** [comment]
So it does recognize them, but still decide to drop them...

**Comment by amitdo** [comment]
Pöllan is dropped because its not in the dictionary and the ö has low certainty
NORDKOREA is dropped because its not in the dictionary and the space has low certainty

**Comment by amitdo** [comment]
USA has the share space with /NORDKOREA but escape from punishment because it's in the dictionary.

**Comment by Shreeshrii** [comment]
@TheSeiko Please try with the changes suggested in https://github.com/tesseract-ocr/tesseract/issues/681#issuecomment-303027906 to see if you get improved recognition of these words without impacting others.

**Comment by amitdo** [comment]
After applying my [patch](https://github.com/tesseract-ocr/tesseract/issues/681#issuecomment-322433100):
'/NORDKOREA' ('USA /NORDKOREA')
and
'Pöllan'

Are recognized in the final text output.

All other words are recognized as before.

**Comment by TheSeiko** [comment]
Thank you to both of you, your help is much appreciated! I'm on holiday till end of next week then I'll try to compile a windows version with the changes you suggested and test it.

**Comment by amitdo** [comment]
>I'll try to compile a windows version with the changes you suggested and test it.

Did you try my suggestion?

**Comment by TheSeiko** [comment]
I'm still on it. 
It takes me some time as I'm used to Java but I only used C/C++/... at University, which is quite some time ago. Then some other projects took my time.
I've already installed Visual Studio 2017 and the Git client.
I'll keep you updated.

**Comment by TheSeiko** [comment]
I've been able to compile it now and starting a test run against 50k frames.

**Comment by TheSeiko** [comment]
looks good

**Comment by TheSeiko** [comment]
Tried it as well with 64bit but there I get some errors but I don't think the problem is the fix:

E:\Tesseract-OCR4.0ab1>tesseract test/1502442621178.png stdout --oem 1 -l Latin
Warning. Invalid resolution 0 dpi. Using 70 instead.
Estimating resolution as 726
Error in pixCreateHeader: height must be > 0
Error in pixCreateNoInit: pixd not made
Error in pixCreate: pixd not made
Error in pixClipRectangle: pixd not made
Error in pixCreateHeader: height must be > 0
Error in pixCreateNoInit: pixd not made
Error in pixCreate: pixd not made
Error in pixClipRectangle: pixd not made
Error in pixCreateHeader: height must be > 0
Error in pixCreateNoInit: pixd not made
Error in pixCreate: pixd not made
Error in pixClipRectangle: pixd not made
Error in pixCreateHeader: height must be > 0
Error in pixCreateNoInit: pixd not made
Error in pixCreate: pixd not made
Error in pixClipRectangle: pixd not made
Error in pixCreateHeader: height must be > 0
Error in pixCreateNoInit: pixd not made
Error in pixCreate: pixd not made
Error in pixClipRectangle: pixd not made
Error in pixCreateHeader: height must be > 0
Error in pixCreateNoInit: pixd not made
Error in pixCreate: pixd not made
Error in pixClipRectangle: pixd not made
Error in pixCreateHeader: height must be > 0
Error in pixCreateNoInit: pixd not made
Error in pixCreate: pixd not made
Error in pixClipRectangle: pixd not made

---
x86 output:
E:\Tesseract-OCR4.0ab1>tesseract test/1502442621178.png stdout --oem 1 -l Latin
Warning. Invalid resolution 0 dpi. Using 70 instead.
Estimating resolution as 726
PATERNION

21-Jähriger geriet mit seinem
Auto in Pöllan zu weit in die
Fahrbahnmitte. Er rammte
das Auto eines 62-Jährigen.

Der Beifahrer des 21-Jährigen
wurde schwer verletzt.

---

I'll try to find the error and keep you updated.

**Comment by amitdo** [comment]
That one is related to image processing. Seems like a bug on (Windows) 64 bit environment.

Please open a new issue for that.

**Comment by Shreeshrii** [comment]
@zdenop Please close this issue.

Words in https://user-images.githubusercontent.com/30631253/29272525-b9762536-8100-11e7-8d82-0961dd49663b.png mentioned above in https://github.com/tesseract-ocr/tesseract/issues/1080#issuecomment-322182649 are not being dropped after patch from @amitdo was merged.

```
PATERNION

21-Jähriger geriet mit seinem
Auto in Pöllan zu weit in die
Fahrbahnmitte. Er rammte
das Auto eines 62-Jährigen.

Der Beifahrer des 21-Jährigen
wurde schwer verletzt.
```

**Comment by amitdo** [comment]
@zdenop, please close this issue.

**Comment by amitdo** [comment]
Fixed in #1264.
